include/bsp/iotc_test_echoserver.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __IOTC_TEST_ECHOSERVER_H__
+#define __IOTC_TEST_ECHOSERVER_H__
+
+#include <netdb.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+const uint16_t kTimeoutSeconds = 1;
+const uint16_t kBufferSize = 1024;
+class EchoTestServer {
+public:
+  EchoTestServer(uint16_t sockType, uint16_t port)
+      : sock_type_(sockType), test_port_(port) {}
+
+  int recv_len_;
+  char recv_buf_[kBufferSize];
+  bool runnable_ = true;
+
+  uint16_t CreateServer();
+  uint16_t RunServer();
+  uint16_t RunTcpServer();
+  uint16_t RunTdpServer();
+  char* get_recv_buf();
+  void stop_server();
```

Comment:
`stop_server()` does not follow the style guide (https://google.github.io/styleguide/cppguide.html#Function_Names)
Please use the `functions should start with a capital letter and have a capital letter for each new word` rule, as you did with the other functions.

For `get_recv_buf()`, it's a special case:
```
Accessors and mutators (get and set functions) may be named like variables. These often correspond to actual member variables, but this is not required. For example, int count() and void set_count(int count).
```
So both `recv_buf()` or `get_recv_buf()` should be acceptable in my opinion.